src/to-have-prop.js

```diff
@@ -18,7 +18,8 @@ export function toHaveProp(element, name, expectedValue) {
   const prop = element.props[name];
 
   const isDefined = expectedValue !== undefined;
-  const isAllowed = VALID_ELEMENTS.includes(element.type);
+  const elementType = typeof element.type == 'string' ? element.type : element.type?.displayName;
```

Comment:
This looks a bit strange, as it will allow any composite component that sets it's displayName to a value from VALID_ELEMENTS list, to have it's props check. AFAIK we want to discourage interaction with composite components in favor of interacting only with host components.
@thymikee, @brunohkbx, wdyt?

Reply:
Yup, I got the same impression, we shouldn't do that.

Reply:
can you explain the logic behind not checking the value of a prop for composite components (or any component)?

Reply:
So the reason why React Native Testing Library ignores composite components, and is only concerned with host components, is that it tries to resemble what user (or manual tester) would see when interacting with the app.  This is similar to React Testing Library, when user interacts with rendered HTML and not composite components.

That said, RNTL already filters out composite components in all queries (getBy, queryBy, findBy), so we probably can ignore checking for host/composite component in `toHaveProp` matchers, and perform prop check on both types. This should not compromise our goal of making it easy to write good test, and hard to write bad tests.

Additionally, this will reduce the "suprise factor" for users trying to validate props on compisite components that they got from using `parent` or `children` props for navigation.

@thymikee, what's your take on that?

Reply:
I'm good with not validating that here. It's the library that's responsible for filtering.

Reply:
did i understand you correctly that the change should be removing the whole `isAllowed` and `VALID_ELEMENTS`
validation?